internal/manifest/manifest.go

```diff
@@ -125,21 +125,53 @@ func (p Policy) Hash() HexString {
 	return NewHexString(hashBytes[:])
 }
 
+// Validate checks the validity of all fields in the manifest.
+func (m *Manifest) Validate() error {
+	for policyHash := range m.Policies {
+		if len(policyHash) != sha256.Size*2 {
+			return fmt.Errorf("policy hash %s has invalid length: %d (expected %d)", policyHash, len(policyHash), sha256.Size*2)
+		} else if _, err := hex.DecodeString(policyHash.String()); err != nil {
+			return fmt.Errorf("decoding policy hash %s: %w", policyHash, err)
+		}
```

Comment:
I'd have a weak preference for decoding first and then checking length vs. `sha256.Size`.